lib/modules/xPostLinks.js

```diff
@@ -0,0 +1,66 @@
+addModule('xPostLinks', function(module, moduleID) {
+	module.moduleName = 'X-post Links';
+	module.category = 'Posts';
+	module.description = 'Create links to x-posted subreddits in the taglines.';
+	module.include = [
+		/^https?:\/\/([a-z]+)\.reddit\.com/i
+	];
+
+	module.go = function() {
+		if ((module.isEnabled()) && (module.isMatchURL())) {
+
+			module.createLinks();
+			window.addEventListener('neverEndingLoad', module.createLinks);
+		}
+	};
+
+	module.createLinks = function () {
+		var
+		titles        = document.querySelectorAll('a.title'),
+		xpost_re      = /x-?post/i,
+		xpost_from_re = /(x-?post\w*\b)(\s+\S+)?\s+(\/?r\/)?([a-zA-Z0-9]+)/i;
+
+		for (var i = 0, title, tagline, entry, userattrs, textNode, subreddit, add_after, xpost_from, xpost_from_a, sub; i < titles.length; i++) {
+
+			title = titles[i];
+
+			if ( xpost_re.test(title.innerHTML) && !title.classList.contains('x-post') ){
```

Comment:
Unless we have a good reason to style xposts, I think it would be better to store "this post has been processed already" as a data attribute ... or not at all.